include/oneapi/dpl/pstl/hetero/dpcpp/unseq_backend_sycl.h

```diff
@@ -542,6 +548,8 @@ struct transform_reduce
             const _Size __adjusted_global_id_plus_one = __adjusted_global_id + 1;
             scalar_reduction_remainder(__adjusted_global_id_plus_one, __adjusted_n, __res, __acc...);
         }
```

Comment:
If I understand correctly, the implicit `else` under this line should actually return false. This is because our `__adjusted_global_id` never processed any work.